File: case_studies/py/vtol_aircraft.py
import numpy as np

def dynamics(t, x, u):
    m, g, Jx, Jy, Jz = 4, 9.81, 0.2, 0.3, 0.4
    x_pos, vx, phi, p, theta, q = x[0], x[1], x[2], x[3], x[4], x[5]
    T = 45  # thrust
    xdot = [vx,
            (T*np.sin(theta) - 0.1*vx) / m,
            p,
            u / Jx,
            q,
            (-T*np.sin(phi)*np.cos(theta) - 0.05*q) / Jy]
    return np.array(xdot)

File: case_studies/m/vtol_aircraft.m
function xdot = dynamics(t, x, u)
    m = 4; g = 9.81; Jx = 0.2; Jy = 0.3; Jz = 0.4;
    x_pos = x(1); vx = x(2); phi = x(3); p = x(4); theta = x(5); q = x(6);
    T = 45; % thrust
    xdot = [vx;
            (T*sin(theta) - 0.1*vx) / m;
            p;
            u / Jx;
            q;
            (-T*sin(phi)*cos(theta) - 0.05*q) / Jy];
end